### DISCUSSION TITLE
How to switch between threads to see their variables and frames?

### DISCUSSION THREAD

**Discussion by vatosarmat** [mfussenegger/nvim-dap#284] [Q&A]
Hello. I am debugging Rust application which does `thread::spawn`. When application stops on a breakpoint I can examine `.scopes` and `.frames` of the current thread. But how to examine `.scopes` and `.frames` of other threads?

**Comment by theHamsta** [comment]
At the moment nvim-dap only supports a single stopped thread. This means that you can currently only inspect the variables of the current thread. dap-ui also fetches a snapshot of the frames other running threads.

Stopping all threads to inspect their variables, would require a refactoring to support multiple stopped threads. Then, the `.pause` command you be used in stopped state to stop additional threads or all threads could be paused automatically after the first one hits a breakpoint.

**Reply by vatosarmat** [reply]
Thanks for reply! Unfortunately, I can't mark it as answer because of some glitch either with my browser or with GitHub frontend 😕

**Comment by drum-computer** [comment]
Hey! I was just wondering, is it still the case in 2024? And if it is, can you please point me in the direction of what has to be changed in the code, I would be interested in contributing this feature potentially.

**Reply by mfussenegger** [reply]
I think even back in 2021 nvim-dap already allowed stopping multiple threads, but it's locked behind a setting because it can be confusing if non-visible threads get blocked by a breakpoint.

You can set:

```lua
require("dap").defaults.fallback.auto_continue_if_many_stopped = false
```

To allow multiple threads to get stopped.

 After that you can switch between the thread's active frame via the threads widget. E.g. either `.threads` in the REPL, expand the thread with `<CR>`, then select the frame with `<CR>` or the same but via `require("dap.ui.widgets").centered_float(require("dap.ui.widgets").threads)`

After the frame is selected, evaluating the variable should show you the right value.
